src/codegen/compile.ml

```diff
@@ -8927,20 +9075,21 @@ and compile_prim_invocation (env : E.t) ae p es at =
   | DerefArrayOffset, [e1; e2] ->
     SR.Vanilla,
     compile_exp_vanilla env ae e1 ^^ (* skewed pointer to array *)
+    Tagged.load_forwarding_pointer env ^^
     compile_exp_vanilla env ae e2 ^^ (* byte offset *)
-    (* Note: the below two lines compile to `i32.add; i32.load offset=9`,
+    (* Note: the below two lines compile to `i32.add; i32.load offset=13`,
        thus together also unskewing the pointer and skipping administrative
        fields, effectively arriving at the desired element *)
     G.i (Binary (Wasm.Values.I32 I32Op.Add)) ^^
-    Arr.load_field 0l                (* loads the element at the byte offset *)
+    Heap.load_field Arr.header_size (* loads the element at the byte offset *)
```

Comment:
Is this to avoid double forwarding in Arr.load_field?

Reply:
Yes, but you are right: It is not consistent with the other places where there is still duplicated forwarding.